taipy/core/_repository/_fs_base.py

```diff
@@ -121,30 +136,62 @@ def _delete_many(self, model_ids: Iterable[str]):
     def _search(self, attribute: str, value: str) -> Optional[Entity]:
         return next(self.__search(attribute, value), None)
 
-    def _get_by_config_and_parent_ids(self, config_id: str, parent_id: Optional[str]) -> Optional[Entity]:
-        for f in self._directory.glob(f"*_{config_id}_*.json"):
-            entity = self.__to_entity(f)
-            if entity.config_id == config_id and entity.parent_id == parent_id:
-                return entity
+    def _get_by_config_and_parent_id(self, config_id: str, parent_id: Optional[str]) -> Optional[Entity]:
+        try:
+            for f in self.dir_path.iterdir():
+                if config_id not in f.name:
+                    continue
+                if entity := self.__to_entity(f, by=parent_id):
+                    if entity.config_id == config_id and entity.parent_id == parent_id:
+                        return entity
+        except FileNotFoundError:
+            pass
         return None
 
+    def _get_by_configs_and_parent_ids(self, config_ids_and_parent_ids):
+        res = {}
+        counter = len(config_ids_and_parent_ids)
+        try:
+            for f in self.dir_path.iterdir():
+                filename = f.name
+                match = [(c, p) for c, p in config_ids_and_parent_ids if c.id in filename]
+                if not match:
+                    continue
+                if entity := self.__to_entity_by(f, by=[p for c, p in match]):
+                    for c, p in match:
+                        if entity.config_id == c.id and entity.parent_id == p:
+                            res[(c, p)] = entity
+                            counter -= 1
+                            if counter == 0:
+                                return res
+                            break
+        except FileNotFoundError:
+            pass
+        return res
+
     def __search(self, attribute: str, value: str) -> Iterator[Entity]:
         return filter(lambda e: getattr(e, attribute, None) == value, self._load_all())
 
-    def __get_model_filepath(self, model_id, raise_if_not_exist=True) -> pathlib.Path:
-        filepath = self._directory / f"{model_id}.json"
-
-        if raise_if_not_exist and not filepath.exists():
-            raise ModelNotFound(str(self._directory), model_id)
+    def __get_model_filepath(self, model_id) -> pathlib.Path:
+        return self.dir_path / f"{model_id}.json"
 
-        return filepath
+    def __to_entity(self, filepath, by=None):
+        with open(filepath, "r") as f:
+            d = f.read()
+        if by and by not in d:
+            return None
```

Comment:
is there a better default value or maybe we can also raise an InvalidModel error

Reply:
In our case, since we are iterating, `None` is simpler.